proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -22631,17 +22730,27 @@ lemma sched_context_bind_tcb_cur_sc_in_release_q_imp_zero_consumed[wp]:
    sched_context_bind_tcb sc_ptr tcb_ptr
    \<lbrace>\<lambda>_ s. cur_sc_in_release_q_imp_zero_consumed s\<rbrace>"
   supply if_split[split del]
-  apply (simp add: sched_context_bind_tcb_def)
+  apply (simp add: sched_context_bind_tcb_def if_cond_refill_unblock_check_def)
 
   \<comment> \<open>split off the setting functions from the start of the definition
       so that sched_context_bind_tcb_cur_sc_in_release_q_imp_zero_consumed_helper can be used\<close>
   apply (subst bind_assoc[symmetric])
   apply (rule hoare_seq_ext[rotated])
    apply (rule sched_context_bind_tcb_cur_sc_in_release_q_imp_zero_consumed_helper)
-
-  apply (wpsimp wp: sched_context_resume_cur_sc_in_release_q_imp_zero_consumed)
-  apply (clarsimp simp: heap_refs_inv_def split: if_splits)
-  apply (rule cur_sc_more_than_ready_imp_cur_sc_ready_and_sufficient; blast?)
+  apply (clarsimp simp: maybeM_def bind_assoc)
+  apply (rule hoare_seq_ext[OF _ get_sched_context_sp])
+  apply (rule hoare_seq_ext[OF _ gets_sp])
+  apply (rule_tac P="K (sc_ptr = xa)" in hoare_pre_tautI; simp)
```

Comment:
`rename_tac` for this `xa`, and there are two `apply`s on the line before `done`